tests/chainer_tests/functions_tests/array_tests/test_permutate.py

```diff
@@ -37,49 +39,55 @@ def check_forward(self, x_data):
                 numpy.testing.assert_array_equal(y_cpu[i], x_data[ind])
 
     def test_forward_cpu(self):
-        self.check_forward(self.x)
+        self.check_forward(self.x, self.indices)
 
     @attr.gpu
     def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x))
+        self.check_forward(cuda.to_gpu(self.x), cuda.to_gpu(self.indices))
 
-    def check_backward(self, x_data, g_data):
-        fun = functions.Permutate(self.indices, axis=self.axis, inv=self.inv)
+    def check_backward(self, x_data, ind_data, g_data):
+        fun = functions.Permutate(axis=self.axis, inv=self.inv)
         gradient_check.check_backward(
-            fun, x_data, g_data)
+            fun, (x_data, ind_data), g_data)
 
     def test_backward_cpu(self):
-        self.check_backward(self.x, self.g)
+        self.check_backward(self.x, self.indices, self.g)
 
     @attr.gpu
     def test_backward_gpu(self):
-        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.g))
+        self.check_backward(cuda.to_gpu(self.x),
+                            cuda.to_gpu(self.indices),
+                            cuda.to_gpu(self.g))
 
 
+@testing.parameterize(
+    {'indices': [0, 0]},
```

Comment:
The original parameter causes different type of error in the new implementation.